### PR TITLE
Add news response model

### PR DESCRIPTION
**Pull request by MariyaSin**
this PR is subtask 231 of tack http://outsourcing.nat.tepkom.ru:20080/redmine/issues/165

Created response mapper to get News. The mapper get DbModel and return the response model, with data about the author and the sender from UserService.

### CODE REVIEW

File: src/NewsService.Mappers/ResponsesMappers/Interfaces/INewsResponseMapper.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+using LT.DigitalOffice.NewsService.Models.Db;
+using LT.DigitalOffice.NewsService.Models.Dto.ModelResponse;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.NewsService.Mappers.ResponsesMappers.Interfaces
+{
+    /// <summary>
+    /// Represents mapper. Provides methods for converting an object of News value <see cref="DbNews"/>
+    /// </summary>
+    public interface INewsResponseMapper : IMapper<DbNews, Task<NewsResponse>>
```
Comment (DarkFelas) [inline src/NewsService.Mappers/ResponsesMappers/Interfaces/INewsResponseMapper.cs:11]: remove Task

File: src/NewsService.Mappers/ResponsesMappers/NewsResponseMapper.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        public async Task<NewsResponse> Map(DbNews value)
+        {
+            if (value == null)
+            {
+                throw new BadRequestException();
+            }
+
+            var authorRequest = IGetUserDataRequest.CreateObj(value.AuthorId);
+
+            var authorResponse = await _client.GetResponse<IOperationResult<IGetUserDataResponse>>(authorRequest);
```
Comment (DarkFelas) [inline src/NewsService.Mappers/ResponsesMappers/NewsResponseMapper.cs:32]: put GetResponse for author and sender in try catch. do not throw exception in catch, just log error

File: src/NewsService.Mappers/ResponsesMappers/NewsResponseMapper.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+using MassTransit;
+using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.NewsService.Mappers.ResponsesMappers
+{
+    public class NewsResponseMapper : INewsResponseMapper
+    {
+        private IRequestClient<IGetUserDataRequest> _client;
+        public NewsResponseMapper(
```
Comment (DarkFelas) [inline src/NewsService.Mappers/ResponsesMappers/NewsResponseMapper.cs:18]: inject logger

**Comment by DarkFelas** [comment]
Do not see task link in description

File: src/NewsService.Mappers/ResponsesMappers/Interfaces/INewsResponseMapper.cs (hunk 1 above)
Comment (MariyaSin) [reply src/NewsService.Mappers/ResponsesMappers/Interfaces/INewsResponseMapper.cs:11]: In the mapper, in the map method, I use asynс/await requests to the userService, which means that I have to return the Task. If I remove the task here, I get an error that I can not use this interface.

File: src/NewsService.Mappers/ResponsesMappers/NewsResponseMapper.cs (hunk 2 above)
Comment (MariyaSin) [reply src/NewsService.Mappers/ResponsesMappers/NewsResponseMapper.cs]: done

File: src/NewsService.Mappers/ResponsesMappers/NewsResponseMapper.cs (hunk 3 above)
Comment (MariyaSin) [reply src/NewsService.Mappers/ResponsesMappers/NewsResponseMapper.cs:21]: done

**Review by TIove — APPROVED** [review]

**Review by borovHokage — APPROVED** [review]

**Pull request merged**